File: PID_with numerical/PID_CNTRL.m
classdef PID_CNTRL
    properties
        Intg_calc
        Diff_calc
        Kp
        Ki
        Kd
    end

    methods
        function self = PID_CNTRL(strt_val,sr)
            self.Intg_calc = Intg(strt_val,sr);
            self.Diff_calc = Diff(strt_val,sr);
            self.Kp = 0.9;  self.Kd = 0.4; self.Ki = 0.2;
        end

        function [self,result] = proc(self,new_e)
            [self.Intg_calc, intg_res] = self.Intg_calc.intg(new_e);
            [self.Diff_calc, dif_res] = self.Diff_calc.diff(new_e);
            res = self.Kd*dif_res + self.Ki*intg_res + self.Kp*new_e;
            result = res;
        end
    end
end

File: PID_NN_wPY/PID/pid_cntrl.py
from numeric_intg import Intg_num
from numeric_dif  import Diff_num

class PID_CNTRL():
    def __init__(self,Kp = 0.9,Kd = 0.4 ,Ki = 0.2 ):
        self.Kp = Kp 
        self.Ki = Ki
        self.Kd = Kd

        self.INTG_BLOCK = Intg_num(1,0)
        self.Diff_num = Diff_num(1,0)

    def _proc(self,new_e):
        intg_res = self.INTG_BLOCK(new_e)
        dif_res  = self.Diff_num(new_e)

        res = self.Kd*dif_res + self.Ki*intg_res + self.Kp*new_e
        
        return res